cloud/external/machineactuator.go

```diff
@@ -66,243 +57,115 @@ const (
-		if ext.v1Alpha1Client != nil {
-			if err := ext.updateAnnotations(cluster, machine); err != nil {
-				glog.Errorf("Could not annotate machine %v: %v\n", machine.ObjectMeta.Name, err)
-				return nil
-			}
-		}
-		glog.Infof("Machine %v created.\n", machine.ObjectMeta.Name)
-
-	} else {
-		glog.Infof("Skipped creating a machine that already exists.\n")
+// Create actuator action powers on the machine
+func (c *ExternalClient) Create(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	// TODO: add some logic that will avoid running fencing command on the macine
+	// that already has desired state
+	glog.Infof("Power-on machine %s", machine.Name)
+	if _, err := c.executeAction(createEventAction, cluster, machine, false); err != nil {
+		glog.Errorf("Could not fence machine %s: %v", machine.Name, err)
+		return c.handleMachineError(machine, errors.CreateMachine(
+			"error power-on instance: %v", err), createEventAction)
 	}
 
+	c.eventRecorder.Eventf(machine, corev1.EventTypeNormal, "Created", "Power-on Machine %s", machine.Name)
+	glog.Infof("Machine %s fencing operation succeeded", machine.Name)
 	return nil
 }
 
-func (ext *ExtClient) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
-	name := machine.ObjectMeta.Name
-
-	// var name string
-	// if machine.ObjectMeta.Annotations != nil {
-	// 	name = machine.ObjectMeta.Annotations[NameAnnotationKey]
-	// }
-
-	instance, err := ext.instanceIfExists(cluster, machine)
-	if err != nil {
-		return err
-	}
+// Delete actuator action powers off the machine
+func (c *ExternalClient) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	// TODO: add some logic that will avoid running fencing command on the macine
+	// that already has desired state
 
-	if instance == nil {
-		glog.Infof("Skipped deleting a machine that is already deleted.\n")
-		return nil
+	glog.Infof("Power-off machine %s", machine.Name)
+	if _, err := c.executeAction(deleteEventAction, cluster, machine, true); err != nil {
+		return c.handleMachineError(machine, errors.DeleteMachine(
+			"error power-off instance: %v", err), deleteEventAction)
 	}
 
-	if _, err := ext.executeAction(deleteEventAction, cluster, machine, true); err != nil {
-		return ext.handleMachineError(machine, errors.DeleteMachine(
-			"error deleting instance: %v", err), deleteEventAction)
-	}
-
-	ext.eventRecorder.Eventf(machine, corev1.EventTypeNormal, "Deleted", "Deleted Machine %v", name)
-
-	return err
-}
-
-func (ext *ExtClient) PostCreate(cluster *clusterv1.Cluster) error {
-	// Ever called?
+	c.eventRecorder.Eventf(machine, corev1.EventTypeNormal, "Deleted", "Power-off Machine %s", machine.Name)
 	return nil
 }
 
-func (ext *ExtClient) PostDelete(cluster *clusterv1.Cluster) error {
-	// Ever called?
+// Update does not run any code
+func (c *ExternalClient) Update(cluster *clusterv1.Cluster, goalMachine *clusterv1.Machine) error {
+	glog.Infof("NOT IMPLEMENTED: update machine %s", goalMachine.Name)
```

Comment:
I'm confused.  You're deleting the implementation and replacing it with a log message?

Reply:
Your implementation, doesn't use it, yet.
The provider isn't just for use by fencing controllers. 
You're being overzealous with these deletions.

Reply:
What do you mean it does not use only for fencing controllers? We gonna use this actuator only for bare metal nodes, if you want to provide an actuator for OSP or oVirt you will need to create additional actuators.

Reply:
Fencing will not be the only client of the Machine API in the system.
The whole point of using the Machine API was to have a single entity that is responsible for all power management, whether it's for fencing, install/scale provisioning, or whatever other use cases people dream up.

In the same way that the fencing logic cannot assume it is talking to this actuator, the actuator cannot assume your fencing logic is one of the entities giving it commands.

Reply:
I did not see that we want to mix OSP, oVirt and bare metal logic in the single actuator, can you please point me where you had such discussion.